vendor/api/v1/authorizenet/views.py

```diff
@@ -84,28 +72,43 @@ def subscription_save_transaction(site, json_data, transaction_detail):
     try:
         payment = subscription.payments.get(transaction=None, status=PurchaseStatus.QUEUED)
         payment.transaction = transaction_id
-        payment.status = PurchaseStatus.CAPTURED
-        payment.result = {}
-        payment.result['raw'] = json_data
+        payment.submitted_date = submitted_datetime
+        payment.result[timezone.now().strftime("%Y-%m-%d_%H:%M:%S")] = payment_info
+        payment.status = payment_status
+        payment.success = payment_success
         payment.save()
         logger.info(f"AuthorizeCaptureAPI subscription_save_transaction: payment {payment.pk} updated")
         
-        processor = AuthorizeNetProcessor(site, payment.invoice)
-        processor.payment = payment
-        processor.create_receipts(payment.invoice.order_items.all())
-        logger.info(f"AuthorizeCaptureAPI subscription_save_transaction: subscription renewed {subscription.pk}")
+        if payment_status == PurchaseStatus.SETTLED:
+            processor = AuthorizeNetProcessor(site, payment.invoice)
+            processor.payment = payment
+            processor.create_receipts(payment.invoice.order_items.all())
+            logger.info(f"AuthorizeCaptureAPI subscription_save_transaction: subscription renewed {subscription.pk}")
 
     except MultipleObjectsReturned as exce:
         # There should be none or only one payment with transaction None and status in Queue
         logger.error(f"AuthorizeCaptureAPI subscription_save_transaction multiple payments returned with None as Transaction, for {subscription_id} exce: {exce}")
         return None
 
     except ObjectDoesNotExist as exce:
+        logger.info(f"AuthorizeCaptureAPI renew_subscription_task subscription {subscription.pk} renewed")
+        invoice = Invoice.objects.create(
+            profile=customer_profile,
+            site=site,
+            ordered_date=submitted_datetime,
+            total=transaction_detail.settleAmount.pyval,
+            status=InvoiceStatus.COMPLETE
+        )
+        invoice.add_offer(offer)
+        invoice.save()
+
+        processor = AuthorizeNetProcessor(site, invoice)
+        processor.subscription = subscription
+        processor.renew_subscription(subscription.gateway_id, payment_info, payment_status, payment_success)
         logger.info(f"AuthorizeCaptureAPI subscription_save_transaction creating new payment and receipt for subscription, for {subscription_id}")
```

Comment:
logging is a little inconsistent on these exceptions. Some are logger.info and others are logger.error. Think they should all be logger.error

Reply:
No does are correct. If `ObjectDoesNotExits` is thrown it means that you are renewing a subscription instead of updating a transaction for a subscription.